### PR TITLE
docs: Document branches and build process for documentation

### PR DESCRIPTION
**Pull request by odzhychko**
Relates to: MODELIX-703

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,3 +21,23 @@ Then, use the included bash script to automatically re-run generation of the doc
 ./runDevelop.sh
 ```
 
+## Source repositories
+
+The documentation is building by using documentation sources from multiple repositories.
```
Comment (languitar) [inline README.md]:
```suggestion
The documentation is built by using documentation sources from multiple repositories.
```

File: README.md [hunk 2]
```diff
@@ -21,3 +21,23 @@ Then, use the included bash script to automatically re-run generation of the doc
...
+You can look at the [playbook](./antora-playbook.yml) to see the up-to-date list of repositories.
+
+##  Versions
+
+The documentation is built for multiple versions.
```
Comment (languitar) [inline README.md]:
```suggestion
The documentation is built for multiple versions of  modelix.
```

File: README.md [hunk 3]
```diff
@@ -21,3 +21,23 @@ Then, use the included bash script to automatically re-run generation of the doc
...
+The latest version named `main` is built from the `HEAD` of the `main` branches of all source repositories.
+An additional version is built for branch matching `docs/release/<version>`.
```
Comment (languitar) [inline README.md]:
```suggestion
An additional version is built for branches matching `docs/release/<version>`.
```

File: README.md [hunk 4]
```diff
@@ -21,3 +21,23 @@ Then, use the included bash script to automatically re-run generation of the doc
...
+
+For example, from the branches (in this and all source repositories) named `docs/release/23.1` a version named `23.1` will be created.
+
+
+### Adding a new version
+
+If you want to add a new version,
+create an appropriately named branch in this repository **and all source repositories**
+pointing to the commit you want the documentation
+to be fetched from.
```
Comment (languitar) [inline README.md]: Maybe one line to match the previous style of [one sentence per line](https://sive.rs/1s)?

**Review by languitar — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (odzhychko) [reply README.md]: Done.

Note: This long sentence was previously broken up using [semantic line breaks](https://sembr.org/).

Comment (languitar) [reply README.md]: sight, yet another convention :D

**Review by languitar — APPROVED** [review]

**Pull request merged**